core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -69,6 +72,9 @@ case class LogConfig(props: java.util.Map[_, _]) extends AbstractConfig(LogConfi
   val minInSyncReplicas = getInt(LogConfig.MinInSyncReplicasProp)
   val compressionType = getString(LogConfig.CompressionTypeProp).toLowerCase
   val preallocate = getBoolean(LogConfig.PreAllocateEnableProp)
+  val messageFormatVersion = ApiVersion(getString(LogConfig.MessageFormatVersionProp))
```

Comment:
Should this be a topic level config? The issue is that this has to match the Api version that the broker uses. For example, if the broker is on 0.9.x, setting the format in a topic to v1 is invalid, but is hard to enforce.